src/app/components/navbar/NavBar.jsx

```diff
@@ -1,12 +1,15 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router";
 import PropTypes from "prop-types";
 import url from "../../utilities/url";
 import auth from "../../utilities/auth";
 import user from "../../utilities/api-clients/user";
+import cookies from "../../utilities/cookies";
 import PreviewNav from "../preview-nav";
 
 const NavBar = props => {
+    const [lang, setLang] = useState(cookies.get("lang"));
```

Comment:
I would be better to read the cookie on component load. 

```
    useEffect(() => {
        setLang(cookies.get("lang"))
    }, []);

```

Reply:
Also I thought this is the change of language for whole florence, maybe you can call it in here 
`const [previewLanguage, setPreviewLanguage] = useState('');`